### PR TITLE
Use apriltags2 for tag detection

### PR DESCRIPTION
**Pull request by henningkayser**
This PR is necessary when using apriltags2 as introduced [here](https://github.com/TAMS-Group/tams_ur5_setup/pull/7). 

The new message type allows multiple poses in a single detection ( bundle detection ). When using single april tag detections as in our configuration, we need to simply choose the first and only pose available in the detection message.

### CODE REVIEW

File: src/camera_positioner.cpp [hunk 1]
```diff
@@ -59,12 +59,12 @@ class CameraPositioner {
       }
    }
 
-   void callback(const apriltags_ros::AprilTagDetectionArray& msg){
+   void callback(const apriltags2_ros::AprilTagDetectionArray& msg){
       // if we got a valid tag detection, update world_camera_transform
       for (int i=0; i< msg.detections.size(); i++) {
-        if(msg.detections[i].id == 0){
+        if(msg.detections[i].id[0] == 0){
```
Comment (v4hn) [inline src/camera_positioner.cpp]: Even if this "can never happen", please check whether `id.size() > 0`.

File: src/camera_positioner_table.cpp [hunk 2]
```diff
@@ -149,11 +149,11 @@ class CameraPositioner {
...
           get_tag0=true;
         }
-        if(msg.detections[i].id == table_tag_id_)
+        if(msg.detections[i].id[0] == table_tag_id_)
 	{
           get_tabletag=true;
-          tabletag_size=msg.detections[i].size;
-         tf::poseMsgToTF(msg.detections[i].pose.pose, tabletag_transform);
+          tabletag_size=msg.detections[i].size[0];
+         tf::poseMsgToTF(msg.detections[i].pose.pose.pose, tabletag_transform);
```
Comment (v4hn) [inline src/camera_positioner_table.cpp]: whitespace error

Comment (v4hn) [reply src/camera_positioner_table.cpp]: Unless you want to add a single commit that does full whitespace-fixup, please fix indentations whenever you touch a line.

Comment (henningkayser) [reply src/camera_positioner_table.cpp]: I fixed that. Both source files somehow have odd indentation. Should I add a commit with unified indentation to this pr?

File: src/camera_positioner.cpp (hunk 1 above)
Comment (henningkayser) [reply src/camera_positioner.cpp]: Fair enough! I wrapped wrapped all that cases by if conditions. Negative cases (should really never happen) are then handled as if no detection was received.

**Comment by v4hn** [comment]
> Negative cases (should really never happen) are then handled as if no detection was received.

Please issue a warning or error in these cases.
The user usually wants to know that some input is not according to specs.
Otherwise you can have a lot of fun looking for this kind of error...

**Pull request closed without merging**